packages/editor/src/store/selectors.js

```diff
@@ -1089,19 +1090,44 @@ export function canUserUseUnfilteredHTML( state ) {
 	] );
 }
 
+/**
+ * Returns whether the given feature is enabled or not.
+ *
+ * @param {Object} state   Global application state.
+ * @param {string} feature Feature slug.
+ *
+ * @return {boolean} Is active.
+ */
+export const isFeatureActive = createRegistrySelector(
+	( select ) => ( state, feature ) => {
+		return !! select( preferencesStore ).get( 'core/edit-post', feature );
+	}
+);
+
+/**
+ * Returns true if the template editing mode is enabled.
+ *
+ * @param {Object} state Global application state.
+ *
+ * @return {boolean} Whether we're editing the template.
+ */
+export const isEditingTemplate = createRegistrySelector(
+	( select ) => ( state ) => {
+		return !! select( EDIT_POST_STORE_NAME ).isEditingTemplate( state );
```

Comment:
The editPost store shouldn't be used in the editor package. 

`select( preferencesStore ).get( 'core/edit-post', feature );` this is also weird (I know it's not introduced in this PR) because it's fetching a preference of edit-post in editor package. cc @talldan 

The reason it shouldn't be used is because it creates a cycle dependency. I'm not sure what you're trying to achieve here but you should probably move your logic to edit-post.

Reply:
@youknowriad When the welcome dialog opens and you select the Close button, focus just kind of gets lost in the void. The goal was to focus the post title field. I thought it might be easiest to fire the focus from the useEffect already defined for post title focus.
1. Create a new post.
2. Welcome dialog appears.
3. Select Close button.
4. Focus is placed in post title field because the post is empty.

This works fine when the welcome dialog does not appear.

Reply:
The PostTitle component though is a generic component that don't assume context (edit-post editor, template mode...). So either we find a generic heuristic and put it in the component or instead move the logic elsewhere (edit-post which has the context about everything)

Reply:
@youknowriad I used a different approach. Hopefully this looks better. Also added a new E2E test so you can see the expected behavior.

Reply:
> select( preferencesStore ).get( 'core/edit-post', feature ); this is also weird (I know it's not introduced in this PR) because it's fetching a preference of edit-post in editor package. cc @talldan

It looks unusual, but I don't think `select( preferencesStore ).get( 'core/editor', feature )` is right. The publish sidebar is part of the post editor and this piece of code says to get whether the post editor's sidebar is enabled. The `core/edit-post` part refers more to the application rather than the package. If a package name like 'core/editor' were used, this becomes seperate from the post editor's other preferences and it becomes hard to know what the preference is for, and whether is should apply to other editors.

Perhaps the issue is that the selector shouldn't be in the `editor` package, it could be deprecated here and moved to the post editor.